tools/ci_connector_ops/ci_connector_ops/pipelines/github.py

```diff
@@ -19,6 +19,8 @@
 def update_commit_status_check(
     sha: str, state: str, target_url: str, description: str, context: str, should_send=True, logger: Logger = None
 ):
+    logger.info(f"Attempting to create {state} status for commit {sha} on Github in {context} context.")
```

Comment:
This call will fail if logger is None, which is the default value.

Reply:
Ahh my mistake.

Question, why do we do this above?
```py
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from logging import Logger
```

Reply:
I'm not sure this is required in this context but sometime we only make imports to use these as type hints. And sometime we can't import a module because of circular dependencies.
`if TYPE_CHECKING` allows to only import a module when `mypy` is performing type checks.